webapi/src/it/scala/org/knora/webapi/e2e/v2/ontology/CardinalitiesV2E2ESpec.scala

```diff
@@ -0,0 +1,560 @@
+          |    {
+          |      "@id" : "$ontologyName:$className",
+          |      "@type" : "owl:Class",
+          |      "rdfs:label" : {
+          |        "@language" : "en",
+          |        "@value" : "$className"
+          |      },
+          |      "rdfs:subClassOf" : {
+          |        "@id" : "$sp"
+          |      }
+          |    }
+          |  ],
+          |  "@context" : {
+          |    "knora-api" : "http://api.knora.org/ontology/knora-api/v2#",
+          |    "$ontologyName" : "$ontologyIri#",
+          |    "owl" : "http://www.w3.org/2002/07/owl#",
+          |    "rdfs" : "http://www.w3.org/2000/01/rdf-schema#",
+          |    "xsd" : "http://www.w3.org/2001/XMLSchema#"
+          |  }
+          |}
+          |""".stripMargin
+    val request = Post(
+      s"$baseApiUrl/v2/ontologies/classes",
+      HttpEntity(RdfMediaTypes.`application/ld+json`, payload)
+    ) ~> addCredentials(rootCredentials)
+    val response = singleAwaitingRequest(request)
+    assert(response.status == StatusCodes.OK, responseToString(response))
+    getLastModificationDate(response)
+  }
+
+  private def createProperty(
+    ontologyIri: String,
+    ontologyName: String,
+    propertyName: String,
+    lastModificationDate: String
+  ) = {
+    val payload =
+      s"""|{
+          |  "@id" : "$ontologyIri",
+          |  "@type" : "owl:Ontology",
+          |  "knora-api:lastModificationDate" : {
+          |    "@type" : "xsd:dateTimeStamp",
+          |    "@value" : "$lastModificationDate"
+          |  },
+          |  "@graph" : [
+          |    {
+          |      "@id" : "$ontologyName:$propertyName",
+          |      "@type" : "owl:ObjectProperty",
+          |      "knora-api:objectType" : {
+          |        "@id" : "knora-api:IntValue"
+          |      },
+          |      "rdfs:label" : {
+          |        "@language" : "en",
+          |        "@value" : "property $propertyName"
+          |      },
+          |      "rdfs:subPropertyOf" : {
+          |        "@id" : "knora-api:hasValue"
+          |      }
+          |    }
+          |  ],
+          |  "@context" : {
+          |    "$ontologyName" : "$ontologyIri#",
+          |    "knora-api" : "http://api.knora.org/ontology/knora-api/v2#",
+          |    "salsah-gui" : "http://api.knora.org/ontology/salsah-gui/v2#",
+          |    "owl" : "http://www.w3.org/2002/07/owl#",
+          |    "rdfs" : "http://www.w3.org/2000/01/rdf-schema#",
+          |    "xsd" : "http://www.w3.org/2001/XMLSchema#"
+          |  }
+          |}
+          |""".stripMargin
+    val request = Post(
+      s"$baseApiUrl/v2/ontologies/properties",
+      HttpEntity(RdfMediaTypes.`application/ld+json`, payload)
+    ) ~> addCredentials(rootCredentials)
+    val response = singleAwaitingRequest(request)
+    assert(response.status == StatusCodes.OK, responseToString(response))
+    getLastModificationDate(response)
+  }
+
+  private def addCardinalityToClass(
```

Comment:
I was thinking to keep the name generic because in terms of what the method does, there is nothing required - it's just how I use it. But I'm happy to change it if you feel it makes it clearer

Reply:
But as far as I understand, the cardinality of "1" is hardcoded in the payload. So, it is not generic and will always be "required".

Reply:
fair enough. Updated the name